bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookKeeperTest.java

```diff
@@ -982,76 +1009,70 @@ public void testEnforceMinNumFaultDomainsForWrite() throws Exception {
-                        sleepLatch.await(10, TimeUnit.SECONDS);
-                        LOG.info("Non default rack bookie awake");
-                    } catch (InterruptedException ignored) {}
+            Thread bookieSleeperCountdown = new Thread(() -> {
+                try {
+                    LOG.info("Counting down 10 seconds before waking sleeping bookie");
+                    sleepLatch.await(10, TimeUnit.SECONDS);
+                    LOG.info("Picked bookie awake");
+                } catch (InterruptedException ignored) {
+                }
 
-                    sleepLatch.countDown();
-                });
+                sleepLatch.countDown();
+            });
 
-                Thread writeToLedger = new Thread(() -> {
-                    try {
-                        LOG.info("Initiating write for entry");
-                        long entryId = lh.addEntry(data);
-                        LOG.info("Wrote entry with entryId = {}", entryId);
-                        assertTrue(entryId >= 0);
-                    } catch (InterruptedException | BKException ignored) {
-                        // Fail the test if the write times out
-                        fail("Write should not have thrown error");
-                    }
-                });
+            Thread writeToLedger = new Thread(() -> {
+                try {
+                    LOG.info("Initiating write for entry");
+                    long entryId = lh.addEntry(data);
+                    LOG.info("Wrote entry with entryId = {}", entryId);
+                    assertTrue(entryId >= 0);
+                } catch (InterruptedException | BKException ignored) {
+                    // Fail the test if the write times out
+                    fail("Write should not have thrown error");
+                }
+            });
 
-                // Putting non default rack bookie to sleep
-                LOG.info("Putting non default rack bookie to sleep");
-                sleepBookie(bookieRack1.get(0), sleepLatch);
+            bookieToSleep = lh.getCurrentEnsemble().get(0);
 
-                // Trying to write entry
-                writeToLedger.start();
+            // Putting non default rack bookie to sleep
+            LOG.info("Putting picked bookie to sleep");
+            sleepBookie(bookieToSleep, sleepLatch);
 
-                // Waiting and checking to make sure that write has not succeeded
-                Thread.sleep(5000);
-                assertTrue("Write succeeded but should not have", writeToLedger.isAlive());
+            // Trying to write entry
+            writeToLedger.start();
 
-                // Starting countdown to wake the bookie
-                bookieSleeperCountdown.start();
-                bookieSleeperCountdown.join();
+            // Waiting and checking to make sure that write has not succeeded
+            Thread.sleep(5000);
+            assertTrue("Write succeeded but should not have", writeToLedger.isAlive());
```

Comment:
it is not appropriate to rely on thread status for validating our logic.

Reply:
Doing.